File: Swift/Problem 1 - 30/Problem 1 - Two Sum.swift
//
//  Problem 1 - Two Sum.swift
//  
//
//  Created by KL on 11/7/2018.
//

/*
    https://leetcode.com/problems/two-sum/description/
*/


class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict: [Int: Int] = [:]
        var i = 0
        while (i < nums.count) {
            let temp = target - nums[i]
            if let index = dict[temp]{
                return [index, i]
            }
            dict[nums[i]] = i
            i += 1
        }
        return []
    }
}

File: Python/Problem 1 - 30/Problem 1 - Two Sum.py
# Your runtime beats 100.00 % of python submissions.
# https://leetcode.com/problems/two-sum/

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}
        for i, num in enumerate(nums):
            k = target - num
            index = dict.get(k)
            if index == None:
                dict[num] = i
            else:
                return [index, i]

        return [2, 3]